core/src/main/java/com/google/errorprone/bugpatterns/FutureReturnValueIgnored.java

```diff
@@ -110,6 +111,11 @@
           instanceMethod()
               .onDescendantOf("io.netty.channel.ChannelFuture")
               .withNameMatching(Pattern.compile("addListeners?")),
+          // ScheduledExecutorService.scheduleAtFixedRate represents a series of tasks where it
+          // doesn't make sense to check for the return value
+          instanceMethod()
+              .onDescendantOf(ScheduledExecutorService.class.getName())
+              .withNameMatching(Pattern.compile("schedule(At|With)Fixed(Rate|Delay)")),
```

Comment:
:thinking: but the `Future` can be used to cancel the scheduling. (Though most users perhaps shut down the whole service instead. I wonder which of those approaches prevails within the Google codebase.)

Reply:
The future can be kept and used for things, of course, but in the specific case of `SES.scheduleAtFixedRate` it's arguably not a mistake to discard the return value. Quoting from the original issue filed:

> The future's get() method will never return normally

So unless you explicitly plan to cancel the future at some point, there's no reason to keep it around. You may want to keep it around to monitor the task, I guess, and restart it if it fails?

Reply:
I left a note in https://github.com/google/error-prone/issues/883#issuecomment-1724432418